From 3de910a19f01a2a9e1c9a6bd6feee1aef547f676 Mon Sep 17 00:00:00 2001
From: Gabor Toth <gabor.toth2@arm.com>
Date: Fri, 3 Mar 2023 12:23:45 +0100
Subject: [PATCH] Update arm_ffa_user driver dependency

Updating arm-ffa-user to v5.0.1 to get the following changes:
 - move to 64 bit direct messages
 - add Linux Kernel v6.1 compatibility
The motivation is to update x-test to depend on the same driver
version as TS uefi-test and thus to enable running these in a single
configuration.
Note: arm_ffa_user.h was copied from:
 - URL:https://git.gitlab.arm.com/linux-arm/linux-trusted-services.git
 - SHA:18e3be71f65a405dfb5d97603ae71b3c11759861

Upstream-Status: Backport

Signed-off-by: Gabor Toth <gabor.toth2@arm.com>
Acked-by: Jens Wiklander <jens.wiklander@linaro.org>
---
 host/xtest/include/uapi/linux/arm_ffa_user.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/host/xtest/include/uapi/linux/arm_ffa_user.h b/host/xtest/include/uapi/linux/arm_ffa_user.h
index 9ef0be3e1664..0acde4fb2020 100644
--- a/host/xtest/include/uapi/linux/arm_ffa_user.h
+++ b/host/xtest/include/uapi/linux/arm_ffa_user.h
@@ -33,7 +33,7 @@ struct ffa_ioctl_ep_desc {
  * @dst_id:	[in] 16-bit ID of destination endpoint.
  */
 struct ffa_ioctl_msg_args {
-	__u32 args[5];
+	__u64 args[5];
 	__u16 dst_id;
 };
 #define FFA_IOC_MSG_SEND	_IOWR(FFA_IOC_MAGIC, FFA_IOC_BASE + 1, \
